architecture/additional_concepts/authentication.adoc

```diff
@@ -345,6 +345,205 @@ desired behavior:
+an "unexpected condition" server error during OAuth, running `oc get events` will display these events under the `ServiceAccount` kind.
+
+The following example warns of a service account that is missing a proper OAuth redirect URI:
+
+----
+$ oc get events | grep ServiceAccount
+1m         1m          1         proxy                    ServiceAccount                                  Warning   NoSAOAuthRedirectURIs   service-account-oauth-client-getter   system:serviceaccount:myproject:proxy has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>
+----
+
+Running oc describe sa/<service-account-name> will also display any OAuth events associated with the given service account name.
+
+----
+oc describe sa/proxy | grep -A5 Events
+Events:
+  FirstSeen     LastSeen        Count   From                                    SubObjectPath   Type            Reason                  Message
+  ---------     --------        -----   ----                                    -------------   --------        ------                  -------
+  3m            3m              1       service-account-oauth-client-getter                     Warning         NoSAOAuthRedirectURIs   system:serviceaccount:myproject:proxy has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>
+----
+
+The following is a  a list of the possible event errors:
+
+**No redirect URI annotations or invalid URI specified**
+
+----
+Reason                  Message 
+NoSAOAuthRedirectURIs   system:serviceaccount:myproject:proxy has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>
+----
+
+**Invalid route specified**
+
+----
+Reason                  Message 
+NoSAOAuthRedirectURIs   [routes.route.openshift.io "<name>" not found, system:serviceaccount:myproject:proxy has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>]
+----
+
+**Invalid reference type specified**
+
+----
+Reason                  Message 
+NoSAOAuthRedirectURIs   [no kind "<name>" is registered for version "v1", system:serviceaccount:myproject:proxy has no redirectURIs; set serviceaccounts.openshift.io/oauth-redirecturi.<some-value>=<redirect> or create a dynamic URI using serviceaccounts.openshift.io/oauth-redirectreference.<some-value>=<reference>]
+----
+
+**Missing SA tokens**
+
+----
+Reason                  Message 
+NoSAOAuthTokens         system:serviceaccount:myproject:proxy has no tokens
+----
+
+===== Sample API Event 
```

Comment:
Sample API Event caused by a possible misconfiguration

I want this section to be "this is a way a user could get into a broken state, and this is how they debug/fix the issue"